File: week6/credit.py
# TODO
from cs50 import get_string

number = get_string("NUMBER: ")
x = number.isnumeric()
while x == False:
    number = get_string("NUMBER: ")
    x = number.isnumeric()
sum = int(0)
i = len(number) - 1
o = int(0)
while i >= 0:
    if o % 2 == 0:
        sum += int(number[i])
    else:
        s = int(number[i])*2
        if s > 9:
            s -= 10
            sum += (s+1)
        else:
            sum += s
    i -= 1
    o += 1
if sum % 10 != 0:
    print("INVALID")
else:
    if number[0] == '3' and o == 15:
        if number[1] in ['4', '7']:
            print("AMEX")
        else:
            print("INVALID")
    elif number[0] == '5' and o == 16:
        if number[1] in ['1', '2', '3', '4', '5']:
            print("MASTERCARD")
        else:
            print("INVALID")
    elif number[0] == '4':
        if o == 16 or o == 13:
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")

File: week1/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // asking the user to enter the number.
    long num = get_long("enter the c.n :");
    int f = 0;
    int s = 0;
    int c = 1;
    int l = 0;
    int g = 0;
    int o = 0;
    int y = 0;
    // checking if the number is valid or not.
    while (num > 0)
    {
        if (c == 16 || c == 15)
        {
            g += num % 10;

            l += num % 10;

            if (c == 16)
            {
                y += num % 10;
            }
        }
        if (c == 14)
        {
            g += num % 10;
        }
        if (c == 13)
        {
            o += num % 10;
        }
        if (c % 2 != 0)
        {
            f += (num % 10);
            num /= 10;
        }
        else
        {
            if (num % 10 >= 5)
            {
                s += (num % 10 - 4 + (num % 10 - 5));
            }
            else
            {
                s += (num % 10) * 2;
            }
            num /= 10;
        }
        c += 1;
    }
    c -= 1;
    s += f;
    if (c == 15)
    {
        if (g == 7 || g == 10)
        {
            if (s % 10 == 0)
            {
                printf("AMEX\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else if (c == 13)
    {
        if (o == 4)
        {
            if (s % 10 == 0)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else if (c == 16)
    {
        if (y == 4)
        {
            if (s % 10 == 0)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else if (l == 6 || l == 7 || l == 8 || l == 9 || l == 10)
        {
            if (s % 10 == 0)
            {
                printf("MASTERCARD\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}